clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CloudConfigRefreshConfig.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.cache;
+
+import com.netflix.spinnaker.cats.cluster.DefaultAgentIntervalProvider;
+import com.netflix.spinnaker.clouddriver.refresh.CloudConfigRefreshScheduler;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.cloud.context.refresh.ContextRefresher;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+/**
+ * Create a {@link CloudConfigRefreshScheduler} to refresh the Spring Cloud Config Server
+ * from an environment repository backend on a schedule that matches the cache refresh
+ * schedule.
+ *
+ * The specified {@code @Profile}s are used by Spring Cloud Config Server to indicate which
+ * environment repository backend(s) are used. Only configure a {@link CloudConfigRefreshScheduler}
+ * if at least one Spring Cloud Config Server profile is active.
+ */
+@Configuration
+@AutoConfigureAfter({RedisCacheConfig.class, DynomiteCacheConfig.class})
+@Profile({"git", "vault", "jdbc", "credhub", "composite"})
```

Comment:
Is there an alternative here to profile naming? (@ConditionalOnBean or something?)

profile naming seems pretty personal to the application configurer and this would be pretty much the first time we've driven behaviour off of it

Reply:
These profiles are what [Spring Cloud uses to decide](https://github.com/spring-cloud/spring-cloud-config/blob/master/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java#L280-L379) which environment repository (or repositories) to configure. It's a bit unfortunate that the profiles don't have names that are more specific to Spring Cloud, but we can't change that. I can see if there's a bean we could depend on instead, but you get most of the Config Server infrastructure with these changes - just with a default `NativeEnvironmentRepository` via the `native` profile without a location configured.

Reply:
Changed conditional to detect bean names instead of profiles.